packages/client/src/authentication/Authentication.ts

```diff
@@ -81,6 +102,33 @@ export class Authentication implements IAuthentication {
     return false;
   }
 
+  async verify(accessToken: string): Promise<boolean> {
+    return this.api.verify(accessToken);
+  }
+
+  async getAccessToken(): PromiseResult<string, CredentialsExpiredError | NotAuthenticatedError> {
+    const credentials = await this.storage.get();
+
+    if (!credentials) {
+      return failure(new NotAuthenticatedError());
+    }
+
+    if (!credentials.shouldRefresh() && credentials.accessToken) {
+      return success(credentials.accessToken);
+    }
+
+    if (credentials.canRefresh()) {
+      const newCredentials = await this.api.refresh(credentials.refreshToken);
+      await this.storage.set(newCredentials);
+
+      invariant(newCredentials.accessToken, 'Access token should be defined');
```

Comment:
It's well possible the refresh token does not get renewed. I would return a
```
return failure(new CredentialsExpiredError())
```
in case of failed refresh, cause it implies the refresh token is expired as well.

Reply:
done!